mlir/test/Dialect/Tensor/invalid.mlir

```diff
@@ -343,20 +343,6 @@ func.func @illegal_collapsing_reshape_mixed_tensor_2(%arg0 : tensor<?x4x5xf32>)
 
 // -----
 
-func.func @expand_shape_invalid_ranks(%arg0: tensor<?x?xf32>) {
-  // expected-error @+1 {{op expected rank expansion, but found source rank 2 >= result rank 2}}
-  %0 = tensor.expand_shape %arg0 [[0], [1]] : tensor<?x?xf32> into tensor<?x?xf32>
```

Comment:
Do we have `invalid.mlir` test cases for `expand_shape` where the result tensor rank is actually reduced? (Same for `collapse_shape`.)

Reply:
I've made an attempt in the change to this PR which I'm about to push